pkg/wakaama/contrib/lwm2m_client_connection.c

```diff
@@ -220,10 +220,10 @@ static int _connection_send(lwm2m_client_connection_t *conn, uint8_t *buffer,
     return 0;
 }
 
-static netif_t *_get_interface(char *host)
+static netif_t *_get_interface(const char *host)
 {
     netif_t *netif = NULL;
-    char *iface = ipv6_addr_split_iface(host);
+    const char *iface = strchr(host, '%') + 1;
```

Comment:
Please use the `zoneid` from the `uri_parser_result_t` here.

Reply:
I still don't see how this requires the new API - wouldn't this just be

```patch
--- a/pkg/wakaama/contrib/lwm2m_client_connection.c
+++ b/pkg/wakaama/contrib/lwm2m_client_connection.c
@@ -220,10 +220,9 @@ static int _connection_send(lwm2m_client_connection_t *conn, uint8_t *buffer,
     return 0;
 }
 
-static netif_t *_get_interface(const char *host)
+static netif_t *_get_interface(const char *iface)
 {
     netif_t *netif = NULL;
-    const char *iface = strchr(host, '%') + 1;
 
     if (iface == NULL) {
         /* get the number of net interfaces */
@@ -330,7 +329,7 @@ static lwm2m_client_connection_t *_connection_create(uint16_t sec_obj_inst_id,
      * if not, check the number of interfaces and default to the first if there
      * is only one defined. */
     if (ipv6_addr_is_link_local((ipv6_addr_t *)&conn->remote.addr.ipv6)) {
-        netif_t *netif = _get_interface(parsed_uri.host);
+        netif_t *netif = _get_interface(parsed_uri.zoneid);
         if (netif == NULL) {
             goto free_out;
         }
```

Reply:
I see you found out, why the new API is required ;-).